app/components/nav-bar.js

```diff
@@ -1,10 +1,22 @@
 import Component from '@ember/component';
+import { computed } from '@ember/object';
 
 export default Component.extend({
   actions: {
     logout() {
       this.authManager.logout();
       this.routing.transitionTo('index');
     }
-  }
+  },
+  nameShown: computed('authManager.currentUser', function() {
```

Comment:
Not a valid fix. This should be corrected at the source of the problem which is the order form. 

`authManager.currentUser` should be considered the source of truth for the current logged in user. the navbar and all other UI elements should always reflect what it contains. 

but the order form on the other hand, _must not_ modify `authManager.currentUser` directly when a user input is still happening. update of `authManager.currentUser` should only happen when the the data on the server itself changes. (ie either buy an update request to the server modifying the user data or after a GET request retrieving the latest user data)

Reply:
@niranjan94 Understood, I will try to formulate another approach . I think I can utilise the `buyer` attribute in the order-form component